proposals/kubeflow-distributions.md

```diff
@@ -0,0 +1,70 @@
+
+## Objective
+
+Clarify how Kubeflow distributions will be owned and developed going forward.
+
+## Motivation
+
+Kubeflow can be divided into pieces
+
+  1. Individual Kubeflow applications (e.g. Pipelines, KFServing, notebooks, etc...)
+  1. Distributions of Kubeflow (e.g. Kubeflow on GCP, Kubeflow on AWS, MiniKF, etc...)
+
+
+ Since July, the Kubeflow community has been working on forming working groups to create greater
+ accountability for the different parts of Kubeflow.
+
+ At this point in time, Kubeflow has formed working groups with clear ownership for all of the individual Kubeflow
+ applications.
+
+ There is an ongoing debate about who should own and maintain Kubeflow distributions.
+
+ To date there are two categories of distributions
+
+  1. Kubeflow distributions tied to a specific platform (e.g. AWS, GCP, etc...)
+  1. Generic distributions (e.g. for MiniKube, any conformant K8s cluster, etc...)
+
+The former have been owned and maintained by the respective vendors. The general consensus is that these should continue
+to be owned and maintained by the respective vendors outside any KF working group.
+
+This leaves the question of what to do about generic distributions. In particular, in [kubeflow/community#402](https://github.com/kubeflow/community/pull/402) there was a long debate about whether the deployments working group would own them or not. That discussion appears to be converging with the decision being that the deployments working group will not own any distributions.
+
+## Proposal
+
+Going forward all distributions of Kubeflow should be owned and maintained outside of Kubeflow. 
```

Comment:
I work with lots of customers trying to adopt Kubeflow in production. One major blocking issue is they cannot easily integrate with entire Kubeflow stack into their existing infrastructure. 
1. Some teams don't own underly k8s and infra team give limited permission to install entire KF stack
2. Some teams have to integrate with their own data models and native Kubernetes solution is not acceptable. They end up build additional wrappers on top of Kubeflow concepts. 

But I have to say, small teams really love upstream distribution and they get an out-of-box ML solution and don't have to put extra efforts on a lot of things.

At the same time, I see a few users use Kubeflow in a different way, they leverage lower level capability from training operators, pipeline API, etc and they pick up  pieces they need from Kubeflow. 

I think these are definitely different directions. Do we want to concentrate on ML on k8s techniques itself? Like how to run distributed training on k8s, how to orchestrate ML workflows on k8s. Or we want to build an e2e ML solution? Then community needs to put efforts on better integration between components, user friendly interfaces, etc.

Reply:
I agree that there is a lot of value in distributions. That's why I think we want to embrace a philosophy of letting a 1000 flowers bloom.

There will be no one distribution that works well or even moderately well for all or even most people. So instead of arguing over what should be in the "generic" distribution I want to encourage folks to organize around use cases, target users and develop distributions optimized for those use cases.

As an example, a distribution targeting someone wanting to kick the tires on their laptop will probably be very different from one running on a small but multi-node cluster.

Reply:
I think I agree on this part. If we see Kubernetes community, it only distribute binaries. There're tons of tools like Minikube, Kind, kops etc to help user provision kubernetes in different envs. The unclear part is even community folks organize their stacks and offer to users, how does Kubeflow community certify them? Or Kubeflow doesn't need to certify them, just say go check list for different solutions